docs/python/tutorials/authentication-authorization.md

```diff
@@ -0,0 +1,127 @@
+---
+sidebar_position: 10
+title: Authentication and Authorization
+description: Use declarative security and authentication middleware in DBOS
+---
+
+DBOS Python supports modular, declarative security.  This is a cooperative effort between the request framework (such as FastAPI) and DBOS Transact; the server framework performs authentication and forwards the authenticated user and roles on to DBOS functions, which then check for authorization.  Authentication information is forwarded via the [DBOS context](../reference/contexts.md#set_authentication).
+
+The following tutorial shows use of FastAPI middleware to collect and authenticate user information, and shows how to set function- and class-level authorization in DBOS.
+
+## Authentication Middleware
+Authentication information may arrive in various ways, depending on the approach, protocol, and framework used.  It is common to use some sort of "middleware" that sits between the request handler and the application code that processes the authentication information in each request.
+
+A simple middleware for FastAPI may pass authentication information to DBOS in a manner similar to this:
```

Comment:
```suggestion
A simple middleware for FastAPI may pass authentication information to DBOS in a manner similar to this. You can construct analogous middleware in Flask or other web frameworks.
```

Reply:
> 

Should we just try it and paste a flask example in below ?

Reply:
Yeah we could tab FastAPI and Flask